VAMobile/src/api/contactInformation/validateAddress.tsx

```diff
@@ -0,0 +1,35 @@
+import { Params as APIParams, post } from 'store/api'
+import { AddressData, AddressValidationData, ValidateAddressData } from 'api/types'
+import { getConfirmedSuggestions, getSuggestedAddresses, getValidationKey } from 'utils/personalInformation'
+import { isErrorObject } from 'utils/common'
+import { logNonFatalErrorToFirebase } from 'utils/analytics'
+import { useMutation } from '@tanstack/react-query'
+
+/**
+ * Validates an address and returns the validation key of the suggested address with the highest confidence score
+ */
+const validateAddress = async (addressData: AddressData): Promise<ValidateAddressData> => {
+  try {
+    const response = await post<AddressValidationData>('/v0/user/addresses/validate', addressData as unknown as APIParams)
+    const suggestedAddresses = getSuggestedAddresses(response)
+    const confirmedSuggestedAddresses = getConfirmedSuggestions(suggestedAddresses)
+    const validationKey = getValidationKey(suggestedAddresses)
+    return { confirmedSuggestedAddresses, suggestedAddresses, validationKey }
+  } catch (error) {
+    throw error
+  }
+}
+
+/**
+ * Returns a mutation for validating an address
+ */
+export const useValidateAddress = () => {
+  return useMutation({
+    mutationFn: validateAddress,
+    onError: async (error) => {
+      if (isErrorObject(error)) {
+        logNonFatalErrorToFirebase(error, 'validateAddress: Service error')
+      }
+    },
+  })
+}
```

Comment:
This could easily be a query, and conceptually it would probably make more sense since this isn't mutating anything. The issue I ran into was handling when to have the query called and passing the data into the query. Because of how the `EditAddressScreen` is set up, even if the query is disabled to run on mount (we only want it to run when a user taps `Save`), the query needs to be manually called with the `refetch` function, but we can't pass parameters for the `queryFn` to the`refetch` function, so we'd need to pass in a dynamic state variable for handling changes to the address from `getAddressValues`. Although it's possible, it turned out to be a little more cumbersome than just having this be a mutation and prop drilling this to the `AddressValidation` component. Might have to think on refactoring this to a query as it'll avoid the need for prop drilling, and we won't have to pass the `mutate` function into the `AddressValidation` component